python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -90,6 +91,24 @@ def get_target_root_asset_partitions(self) -> Iterable[AssetKeyPartitionKey]:
             self.target_subset.filter_asset_keys(root_asset_keys).iterate_asset_partitions()
         )
 
+    def get_target_root_partitions_subset(self) -> Optional[PartitionsSubset]:
+        root_asset_keys = (
+            AssetSelection.keys(*self.target_subset.asset_keys)
+            .sources()
+            .resolve(self.target_subset.asset_graph)
+        )
+
+        if (
+            not root_asset_keys
+            or self.target_subset.asset_graph.get_partitions_def(next(iter(root_asset_keys)))
+            is None
+        ):
+            return None
+
+        # An asset backfill must target the same partitions for the root assets
+        # Return the partitions subset for the first root asset
+        return self.target_subset.get_partitions_subset(next(iter(root_asset_keys)))
```

Comment:
What if one of the roots isn't partitioned but the others are?

Reply:
I thought we only allow creating an asset backfill when the roots share the same partitioning?

https://github.com/dagster-io/dagster/blob/607478333d44788b4063de708a9a0ea01a2a6ec9/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L217

Reply:
That's after excluding non-partitioned assets (look at the lines directly above that link)

Reply:
Ok, I've updated this.

I think it's possible that a backfill can still be created when the upstream asset is unpartitioned and the downstream assets are partitioned. I think in that case, we want to display the partition ranges of the most upstream partitioned asset, which is what I've amended this PR to do. 

I think in all cases there should be at least one partitioned asset in the targeted backfill, let me know if this is incorrect @sryza